### ISSUE TITLE
`Image` component don't care about updating source from component outside 

### ISSUE THREAD

**Issue by CHNB128** [GeekyAnts/NativeBase#4275]
** Before reporting this bug, please...**

1. Check the latest documentation: https://docs.nativebase.io/
2. Check for existing open/closed issues for a possible duplicate before creating a new issue: https://github.com/GeekyAnts/NativeBase/issues
3. Use the latest NativeBase release: https://github.com/GeekyAnts/NativeBase/releases
4. Check examples from NativeBase KitchenSink https://github.com/GeekyAnts/NativeBase-KitchenSink
5. For discussion purpose make use of NativeBase discord: https://discord.gg/TSgCw2UPmb

**Describe the bug**
`Image` component don't care about updating source from component outside 

**To Reproduce**
Update `source` or `src` of `Image`, will see it will stay static.

**Expected behaviour**
Source should handle external variable change

**CodeSandBox/Snack link**
https://codesandbox.io/s/pedantic-christian-lbhkx?file=/src/components/Example.tsx

**Screenshots**

**Platform**

- NativeBase version: 3.2.1
- Device: Any
- OS: Any
- Browser Any
- Browser version Any
- React Native version 0.64.2
- Expo sdk version 43
- node 14.16.1
- yarn 1.22.15

**Additional information**

1. Add any other context about the problem here.
2. Is the bug present in both iOS and Android or in any one of them?
3. If you want your issue to be looked at quicker, please attach a snack reproducible with your issue. Makes it easier for us!

**Comment by ocarreterom** [comment]
This seems to be a problem on react-native `Image` component. Adding a `key` should fix that.
https://github.com/facebook/react-native/issues/9195#issuecomment-238190537

**Comment by MD-REHMAN** [comment]
Thanks @ocarreterom, and we'll keep this issue open till [react-native/image](https://github.com/facebook/react-native/issues/9195#issuecomment-238190537) get resolved.

**Comment by Leleka14** [comment]
for me it forked to randomly assign imageIndex only once when component is mounted and if image is changed, it only takes to refresh screen, may not be the best solution, but at least image is not flickering

const randomValue = useMemo(()=>Math.random(), [])

<Image src={{uri: https://image/image.jpg?id={randomValue}}}/>
